assets/js/blocks/product-gallery/view.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import { store as interactivityApiStore } from '@woocommerce/interactivity';
+
+interface State {
+	[ key: string ]: unknown;
+}
+
+interface Context {
+	productGallery: { numberOfThumbnailImages: number };
+}
+
+interface Selectors {
+	productGallery: {
+		getNumberOfPages: ( store: unknown ) => number;
+	};
+}
+
+interface Store {
+	state: State;
+	context: Context;
+	selectors: Selectors;
+	ref: HTMLElement;
+}
+
+type SelectorsStore = Pick< Store, 'context' | 'selectors' >;
+
+interactivityApiStore( {
+	selectors: {
+		productGallery: {
+			getNumberOfPages: ( store: SelectorsStore ) => {
+				const { context } = store;
+
+				return context.productGallery.numberOfThumbnailImages;
```

Comment:
I think I'd simplify the name to:
```suggestion
				return context.productGallery.numberOfThumbnails;
```

What do you think?

Reply:
I think your suggestion is better! I just renamed it. Thank you!